.github/workflows/ci.yml

```diff
@@ -0,0 +1,28 @@
+name: Continuous Integration
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+permissions: read-all
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+    permissions: read-all
+    strategy:
+      fail-fast: false
+      matrix:
+        # node 8 is technically supported but it is quite difficult to get it to build correctly these days
```

Comment:
What's the issue with v8? What about v12+?

Reply:
I added `12` and `14` to the test matrix but it looks like we'd some work to support these.

Reply:
Looks like the tests are just flaky? I'll restructure them and check again on Monday